microcosm_flask/conventions/config.py

```diff
@@ -0,0 +1,83 @@
+"""
+Config sharing convention.
+
+Reports service config protected by basic auth for securely running services
+locally with realistic config.
+
+"""
+from distutils.util import strtobool
+from microcosm.api import defaults
+from json import dumps, loads
+
+from microcosm_flask.audit import skip_logging
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.conventions.encoding import make_response
+from microcosm_flask.namespaces import Namespace
+from microcosm_flask.operations import Operation
+
+
+class Config:
+    """
+    Wrapper around service config state.
+
+    """
+    def __init__(self, graph, include_build_info=True):
+        self.graph = graph
+        self.name = graph.metadata.name
+        self.config = dict(
+            self.graph.config
+        )
+
+    def to_dict(self):
+        """
+        Encode the name, the status of all checks, and the current overall status.
+
+        """
+        def remove_nulls(dct):
```

Comment:
How come? In theory null values are meaningful.

Reply:
Well the real problem is that things that were not json in the original dump are converted to `null` before passed off to flask so `null` might mean "some object that was automatically loaded in". Not cleaning out the nulls would mean overridden config loaded within the code--like when we configure the PubsubEnvelope for a daemon and default it to the SQSEnvelope.